### ISSUE TITLE
is_catalog_frontpage() returns true on featured collection

### ISSUE THREAD

**Issue by clifgriffin** [ingenesis/shopp#3598]
Bug in 1.3.11:

When viewing featured collection, `is_catalog_frontpage()` returns true, causing pagination links to break because of this line in `ProductCollection:pagelink()`:

``` php
if ( empty($prettyurl) && is_catalog_frontpage() ) $prettyurl = ($page > 1 || $alpha ? "page/$page" : "");
```

**Comment by Just-m** [comment]
Tried to replicate the issue. Is it already solved in 1.3.12.? If not in what way does it break the pagination link(s).

Are you refering to this line https://github.com/ingenesis/shopp/blame/1.3.x/core/model/Collection.php#L278

```php
if ( is_catalog_frontpage() ) $prettyurl = ($page > 1 || $alpha ? "page/$page" : "");
```

it should say
```php
if ( is_shopp_catalog_frontpage() ) $prettyurl = ($page > 1 || $alpha ? "page/$page" : "");
```
But other than that..., it seems it is last edited 4 years ago, so I am not sure this is the right line.

**Comment by clifgriffin** [comment]
I need to do a better job of documenting these things.

Did you test whether `is_shopp_catalog_frontpage()` returns true on the featured collection?

**Comment by Just-m** [comment]
Yes it is returning `true`. It is doing that for all smartcategories.
Which is probably correct, considering the smartcategories are virtual categories. Just the catalog view with a filter applied.

What I don't see is how, when, where, it breaks pagination (links). It seems to work fine.

**Comment by Just-m** [comment]
This function probably doesn't do what you think it should does.
In the function 
```
function is_shopp_catalog_frontpage () {
	$Page = ShoppPages()->requested();
	return $Page !== false && $Page->name() == 'catalog';
}
```
uses

```
	public function requested () {
		if ( ! isset($this->slugs[ ShoppPages::request() ]) ) return false;
		$pagename = $this->slugs[ ShoppPages::request() ];
		return $this->get( $pagename );
	}
```
These slugs only relate to the default Shopp pages:
========= slugs in flow/Pages ============

Array
(
[shop] => catalog
[account] => account
[cart] => cart
[checkout] => checkout
[confirm-order] => confirm
[thanks] => thanks
)

`ShoppPages->request()` returns 'shop', so the function `requested()` returns 'catalog', making it the shopp-catalog-frontpage.

**Comment by clifgriffin** [comment]
@Just-m You may be right. If this is a feature, then we have a problem I think. The name `is_shopp_catalog_frontpage()` suggests to me this should ONLY return true on the actual catalog front page, i.e. `/shop`. 

We have these for the general evaluation:
https://shopplugin.net/api/is_shopp_page/
https://shopplugin.net/api/is_catalog_page/

Pretty sure that last one is deprecated.

**Comment by Just-m** [comment]
Yes, it doesn't work the way described in the API docs.

**Comment by Just-m** [comment]
My logic doesn't work for all categories so it seems, eventhough they are also loaded into the main Shopp page. Changing the code to this 

```php
function is_shopp_catalog_frontpage ( $wp_query = false ) {
	$Page = ShoppPages()->requested();
	return $Page !== false && $Page->name() == 'catalog' && ! is_shopp_smart_collection();
} 
```
makes it work, in the way that it only returns `true` when you access the catalog products using a link to the shop page ('www.yourdomain.com/shop/'). 
On a page like 'www.yourdomain.com/shop/collection/catalog/', which sort of displays the same information, it will return `false`.

**Comment by Just-m** [comment]
Fixed #3661
